flow/src/main/scala/org/alephium/flow/core/TxUtils.scala

```diff
@@ -452,6 +452,152 @@ trait TxUtils { Self: FlowUtils =>
     }
   }
 
+  protected[core] def buildChangeTxAndGetAvailableInputs(
+      fromLockupScript: LockupScript.Asset,
+      fromUnlockScript: UnlockScript,
+      availableInputs: AVector[AssetOutputInfo],
+      outputGroups: AVector[AVector[TxOutputInfo]],
+      gasPrice: GasPrice
+  ): Either[String, (UnsignedTransaction, AVector[AssetOutputInfo])] = {
+    val totalOutputsCount = outputGroups.map(_.length).sum
+    if (totalOutputsCount <= availableInputs.length) {
+      Left("Change transaction is not needed as there is enough inputs")
+    } else {
+      val highestValueUtxoToBreakDown = availableInputs.maxBy(_.output.amount)
+      val missingUtxoCount            = totalOutputsCount - availableInputs.length
+      val changeUtxoAmount =
+        highestValueUtxoToBreakDown.output.amount.divUnsafe(U256.unsafe(missingUtxoCount + 1))
+      val breakDownUtxos = AVector.fill(missingUtxoCount)(
+        TxOutputInfo(
+          fromLockupScript,
+          changeUtxoAmount,
+          AVector.empty,
+          None
+        )
+      )
+      val inputs = AVector(highestValueUtxoToBreakDown.ref -> highestValueUtxoToBreakDown.output)
+
+      for {
+        gasNeeded <-
+          GasEstimation.estimateWithInputScript(
+            fromUnlockScript,
+            inputs.length,
+            breakDownUtxos.length + 1,             // + 1 is for change utxo
+            AssetScriptGasEstimator.NotImplemented // Not P2SH
+          )
+        tx <- UnsignedTransaction
+          .buildTransferTx(
+            fromLockupScript,
+            fromUnlockScript,
+            inputs,
+            breakDownUtxos,
+            gasNeeded,
+            gasPrice
+          )
+      } yield {
+        val changeUtxos = tx.fixedOutputs
+          .mapWithIndex { case (output, outputIndex) =>
+            AssetOutputRef.from(output, TxOutputRef.key(tx.id, outputIndex)) -> output
+          }
+          .collect {
+            case (ref, output) if output.lockupScript == fromLockupScript =>
+              AssetOutputInfo(ref, output, MemPoolOutput)
+          }
+        (tx, availableInputs.filterNot(_ == highestValueUtxoToBreakDown) ++ changeUtxos)
+      }
+    }
+  }
```

Comment:
This method exists for making and getting change utxos that we needed for making a transfer to more destinations that available utxos.